lxd/db/node.go

```diff
@@ -1190,6 +1190,99 @@ func (c *ClusterTx) GetMemberConfigWithGlobalDefault(ctx context.Context, cluste
 	return rows, err
 }
 
+// InstanceKey maps instance name and config key to optional value.
+type InstanceKey struct {
+	InstanceName string
+	Key          string
+	Value        sql.NullString
+}
+
+// GetClusterMemberInstanceConfig returns a map from clusterMemberName -> InstanceKey
+// with optional filters clusterMemberNames and keys. If either slice is nil/empty,
+// the results will include all cluster members/config keys.
+func (c *ClusterTx) GetClusterMemberInstanceConfig(ctx context.Context, clusterMemberNames []string, keys []string) (map[string][]InstanceKey, error) {
```

Comment:
This doesnt look workable to me because your not expanding the effective instance config from profile(s).

I dont think you can do this in a single query, you might need to use `InstanceList()` instead with a custom handler func.

Reply:
Focusing on this as it's the largest functional deficiency; the current commits are a bit of a mess but my feeling is that  `InstanceList` won't quite work here. In order to add an instance (with expanded config) we need a list of all profiles that will be applied to that instance, which we can't get from `InstanceList`. Still thinking on this...

Reply:
You can use InstanceList for this, we do it commonly. It doesn't expand it automatically but does provide profile info so you can expand if needed. See existing examples

Reply:
e.g.

```
err = s.DB.Cluster.Transaction(context.TODO(), func(ctx context.Context, tx *db.ClusterTx) error {
		return tx.InstanceList(ctx, func(dbInst db.InstanceArgs, p api.Project) error {
                    dbInst.Config = instancetype.ExpandInstanceConfig(globalConfigDump, dbInst.Config, dbInst.Profiles)
			return nil
		}, filter)
	})
	if err != nil {
		return nil, err
	}

```

Reply:
im confused, can you book a quick meeting to discuss as i dont see what the issue is.